src/cmd.c

```diff
@@ -1089,6 +1090,139 @@ static void pctx(atci_param_t *param)
     request_confirmation = true;
 }
 
+static void cw(atci_param_t *param)
+{
+    uint32_t freq, timeout;
+    int32_t power;
+
+    if (!atci_param_get_uint(param, &freq)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_int(param, &power)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_uint(param, &timeout)) abort(ERR_PARAM);
+
+    log_debug("$CW: freq=%ld power=%ld timeout=%ld\r\n",  freq, power, timeout);
+
+    MlmeReq_t mlr = { .Type = MLME_TXCW };
+    mlr.Req.TxCw.Timeout = timeout;
+    mlr.Req.TxCw.Frequency = freq;
+    mlr.Req.TxCw.Power = power;
+
+    abort_on_error(LoRaMacMlmeRequest(&mlr));
+
+    OK_();
+}
+
+
+static void SX1276SetOpMode( uint8_t opMode )
+{
+    if( opMode == RF_OPMODE_SLEEP )
+    {
+        SX1276SetAntSwLowPower( true );
+    }
+    else
+    {
+        // Enable TCXO if operating mode different from SLEEP.
+        SX1276SetBoardTcxo( true );
+        SX1276SetAntSwLowPower( false );
+        SX1276SetAntSw( opMode );
+    }
+    SX1276Write( REG_OPMODE, ( SX1276Read( REG_OPMODE ) & RF_OPMODE_MASK ) | opMode );
+}
+
+
+static void cm_clk_irq_handler( void* context )
+{
+    (void) context;
+
+    static uint32_t i = 0;
+    i++;
+    gpio_write(RADIO_DIO_2_PORT, RADIO_DIO_2_PIN, (i % 2) == 0);
+
+    /*
+        static bool pin = true;
+
+        // toggle every second when datarate is 4800
+        if(i % 4800 == 0)
+        {
+            atci_printf("X");
+            pin = !pin;
+        }
+
+        gpio_write(RADIO_DIO_2_PORT, RADIO_DIO_2_PIN, pin);
+
+    */
+}
+
+
+static void cm(atci_param_t *param)
+{
+    (void)param;
+
+    // Example preamble 5, datarate 4800
+    // AT$CM 868300000,250000,4800,5,-10,3
+    uint32_t freq, timeout, fdev, datarate, preamble;
+    int32_t power;
+
+    if (!atci_param_get_uint(param, &freq)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_uint(param, &fdev)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_uint(param, &datarate)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_uint(param, &preamble)) abort(ERR_PARAM);
```

Comment:
How about generalizing this parameter slightly so that the testing lab or user could specify an arbitrary bit stream to be transmitted continuously? We could switch the parameter to a hexadecimal format and let the caller specify an arbitrarily long bit stream, e.g.,
```
AT$CM 868300000,250000,4800,-10,3,0123456789ABCDEF
```
Then we could support arbitrary bit streams, including randomly generated data.

In the above example I'm also switching the order of parameters so that the the variable-length parameter comes at the end of the parameter list, but that's not really necessary (just more common).

For higher modulation data rates or longer bit streams, we could also consider using the same AT command syntax that commands like AT+UTX use, where the data to be transmitted has explicit length and comes on a separate line.

Reply:
This might happen if the testing lab would need some pseudo random binary sequence. I would keep that for another PR.

Creating commands for custom radio packets is doable however, most stuff I use at home has OOK modulation at 433 Mhz. Not sure what we can do with FSK at 868 MHz (because of 868 MHz antenna & filters inside Murata). Maybe simulating meteo stations, some of them use 868 FSK.